src/battle_terastal.c

```diff
@@ -56,33 +64,31 @@ bool32 CanTerastallize(u32 battler)
     u32 holdEffect = GetBattlerHoldEffect(battler, FALSE);
 
     // Check if Player has Tera Orb and has charge.
-    if (!CheckBagHasItem(ITEM_TERA_ORB, 1)
-        || !((B_FLAG_TERA_ORB_NO_COST != 0 && FlagGet(B_FLAG_TERA_ORB_NO_COST))
-        || (B_FLAG_TERA_ORB_CHARGED != 0 && FlagGet(B_FLAG_TERA_ORB_CHARGED)
-        && ((battler == B_POSITION_PLAYER_LEFT || (!(gBattleTypeFlags & BATTLE_TYPE_MULTI) && battler == B_POSITION_PLAYER_RIGHT))))))
+    if (!TESTING && !(B_FLAG_TERA_ORB_NO_COST != 0 && FlagGet(B_FLAG_TERA_ORB_NO_COST))
+        && (battler == B_POSITION_PLAYER_LEFT
+        || (!(gBattleTypeFlags & BATTLE_TYPE_MULTI) && battler == B_POSITION_PLAYER_RIGHT)))
     {
-        return FALSE;
+        if (!CheckBagHasItem(ITEM_TERA_ORB, 1))
+            return FALSE;
+        if (B_FLAG_TERA_ORB_CHARGED != 0 && !FlagGet(B_FLAG_TERA_ORB_CHARGED))
+            return FALSE;
```

Comment:
Suggestion 
```
    if (!CheckBagHasItem(ITEM_TERA_ORB, 1))
        return FALSE;
    
`    // Check if Player has Tera Orb and has charge.
    if (!TESTING 
     && !(B_FLAG_TERA_ORB_NO_COST != 0 && FlagGet(B_FLAG_TERA_ORB_NO_COST))
     && (battler == B_POSITION_PLAYER_LEFT || (!(gBattleTypeFlags & BATTLE_TYPE_MULTI) && battler == B_POSITION_PLAYER_RIGHT)))
    {

        if (B_FLAG_TERA_ORB_CHARGED != 0 && !FlagGet(B_FLAG_TERA_ORB_CHARGED))
            return FALSE;
    }
```
Since it will be always false if the tera orb is present I would just skip all the calcs and the other suggestion is some small formatting to make it obvious that the or condition belongs to the and condition.